src/error.rs

```diff
@@ -8,7 +9,7 @@ use thiserror::Error;
 /// them. This is especially true for calls that involve compiling raw binary data to WebAssembly
 /// modules. Because of this an opaque error ID is returned from host that can be transformed to
 /// a string.
-#[derive(Error)]
+#[derive(Error, Serialize, Deserialize)]
```

Comment:
It's not safe to serialise errors, they are also resources and bound the the running process.